backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -136,6 +138,33 @@ public List<FolderResponse> getAllFoldersWithParent(UUID parentFolderId) {
     return asFolderResponseList(allFolders);
   }
 
+  @Override
+  public void moveFolderToBucket(UUID folderId) {
+    FolderEntity folderEntity =
+        folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+    if (folderEntity.getParentFolder() == null) {
+      throw new FolderAlreadyInsideBucketException();
```

Comment:
why not throw FolderAlreadyInsideFolder here?

Reply:
because folder is not really inside any folder here :)